provider/provider.go

```diff
@@ -0,0 +1,32 @@
+package provider
+
+import (
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/terraform"
+)
+
+func New(seed *schema.Provider) *Provider {
+	return &Provider{
+		provider: seed,
+	}
+}
+
+type Provider struct {
```

Comment:
Question: `Provider.provider` seems weird.  Are we able to do something such as: `type Provider *schema.Provider` instead?  Maybe this isn't possible :)

Reply:
Thanks for chatting with me about it and letting me bounce thoughts off you. Updated in https://github.com/terraform-providers/terraform-provider-vault/pull/737/commits/5fb25abe536f80b0965eb3f251b96f27cd889ca4.